packages/jest-leak-detector/src/index.ts

```diff
@@ -48,6 +48,17 @@ export default class LeakDetector {
       await tick();
     }
 
+    if (this._isReferenceBeingHeld) {
```

Comment:
this adds a full 300ms to the runtime of the simplest test I can manage (on a beefy macbook pro). But doing it here means we only have the hit if we believe we're still leaking after running GC, so I think it's worthwhile